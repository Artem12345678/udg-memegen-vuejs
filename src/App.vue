<template>
  <div id="app">
    <Header/>

    <main class="container">
      <router-view/>
    </main>

    <Footer/>
  </div>
</template>

<script>
import Header from './components/Header/Header.vue';
import Footer from './components/Footer/Footer.vue';

export default {
  name: 'App',
  components: {
    Header,
    Footer,
  },
};
</script>

<style lang="scss">
  #app {
    min-height: 100vh;
    display: flex;
    flex-direction: column;
  }

  main {
    flex-grow: 1;
  }
</style>
