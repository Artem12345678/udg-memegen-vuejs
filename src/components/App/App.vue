<template>
  <div id="app">
    <AppHeader />

    <main class="container">
      <router-view />
    </main>

    <AppFooter />
  </div>
</template>

<script>
import AppHeader from '@/components/AppHeader/AppHeader.vue';
import AppFooter from '@/components/AppFooter/AppFooter.vue';

export default {
  name: 'App',
  components: {
    AppHeader,
    AppFooter,
  },
};
</script>

<style lang="scss">
#app {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}

main {
  flex-grow: 1;
}
</style>
