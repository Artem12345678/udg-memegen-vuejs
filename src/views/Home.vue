<template>
  <div class="home">
    <MemeGen />
  </div>
</template>

<script>
import MemeGen from '@/components/MemeGen/MemeGen.vue';

export default {
  name: 'Home',
  components: {
    MemeGen,
  },
};
</script>
