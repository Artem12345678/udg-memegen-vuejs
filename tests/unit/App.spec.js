import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import VueRouter from 'vue-router';
import { BootstrapVue } from 'bootstrap-vue';
import Verte from 'verte';

import App from '@/components/App/App.vue';
import AppHeader from '@/components/AppHeader/AppHeader.vue';
import AppFooter from '@/components/AppFooter/AppFooter.vue';
import Home from '@/views/Home.vue';

import { routes } from '@/router';

const localVue = createLocalVue();

localVue.use(VueRouter);
localVue.use(BootstrapVue);

localVue.component('verte', Verte);

const router = new VueRouter({
  routes,
});

describe('App.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(App, {
      localVue,
      router,
      sync: false,
    });
  });

  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders <AppHeader/> component', () => {
    expect(wrapper.find(AppHeader).exists()).toBe(true);
  });

  it('renders <AppFooter/> component', () => {
    expect(wrapper.find(AppFooter).exists()).toBe(true);
  });

  describe('with router', () => {
    beforeEach(() => {
      wrapper = mount(App, {
        localVue,
        router,
        sync: false,
      });
    });

    it('renders <Home/> component by default', () => {
      expect(wrapper.find(Home).exists()).toBe(true);
    });

    it('renders <Home/> component for "/" route path', async () => {
      await router.push('/123');
      await router.push('/');

      await wrapper.vm.$nextTick();
      expect(wrapper.find(Home).exists()).toBe(true);
    });

    it('renders <Home/> component for "home" route name', async () => {
      await router.push('/123');
      await router.push({ name: 'home' });

      await wrapper.vm.$nextTick();

      expect(wrapper.find(Home).exists()).toBe(true);
    });
  });
});
