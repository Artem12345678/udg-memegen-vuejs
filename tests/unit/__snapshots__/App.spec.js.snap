// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches snapshot 1`] = `
<div
  id="app"
>
  <appheader-stub />
   
  <main
    class="container"
  >
    <router-view-stub
      name="default"
    />
  </main>
   
  <appfooter-stub />
</div>
`;
